workflow/installertest.groovy

```diff
@@ -0,0 +1,133 @@
+// Replace colons in image with hyphens and append text
+String convertImageNameToString(String imageName, String append="") {
+    return (imageName+append).replaceAll(':','-')
+}
+
+/**
+ * Extracts the components from an {@code 'artifact://full/path/to/job/buildNr#artifact.ext'} type url.
+ * @param url the url
+ */
+@NonCPS
+def getComponentsFromArtifactUrl(String url) {
+    def pattern = /^artifact:\/\/([\/\w-_ ]+)\/(\d+)\/{0,1}#([\/\w\.]+)$/
+    def matcher = url =~ pattern
+    if (matcher) {
+        return [
+            item: matcher.group(1),
+            run: matcher.group(2),
+            artifact: matcher.group(3)
+        ]
+    } else {
+        throw new MalformedURLException("Expected format: 'artifact://full/path/to/job/buildNr#dir/artifact.ext' but got '${url}'")
+    }
+}
+
+/**
+ * Downloads an arifact to the workspace for further use in the flow.
+ * This is a stripped down version of a CloudBees workflow utility function
+ * It has been generified to use with artifacts besides jenkins.war (for packaging use)
+ * 
+ * Caveats:
+ * Needs to run inside a node block, and does not stash the artifact (avoids issues with big artifacts)
+ * For smaller artifacts, you may wish to stash them
+ * @param url the url to download
+ *              {@code 'artifact://' will be downloaded via CopyArtifact build step},
+ *              Example: {@code 'artifact://full/path/to/job/buildNr#artifact.ext'}.
+ *              Anything else will be downloaded via wget
+ */
+def fetch_artifact(String url) {
+  if (url == null) {
+    fail 'required parameter url is missing'
+  } else if (url.startsWith("artifact://")) {
+    echo "Fetching ${url} as artifact."
+    def comp = this.getComponentsFromArtifactUrl(url)
+    step([$class: 'CopyArtifact', filter: comp.artifact, projectName: comp.item, selector: [$class: 'SpecificBuildSelector', buildNumber: comp.run]])
```

Comment:
this should really have `flatten:true` added, else you get the full path to the artifact.